dbt-tests-adapter/src/dbt/tests/adapter/utils/test_current_timestamp.py

```diff
@@ -47,8 +47,11 @@ def utcnow_matching_type(self, dt: datetime) -> datetime:
         """
         Current UTC datetime with the same timezone-awareness (or naiveness) as the input.
         """
-        return datetime.now(timezone.utc) if is_aware(dt) else datetime.utcnow()
-
+        return (
+            datetime.now(timezone.utc)
+            if is_aware(dt)
+            else datetime.now(timezone.utc).replace(tzinfo=None)
+        )
```

Comment:
```suggestion
        if is_aware(dt):
            return datetime.now(timezone.utc)

        return datetime.now(timezone.utc).replace(tzinfo=None)
```

Other options considered:

```python
        if not is_aware(dt):
            return datetime.now(timezone.utc).replace(tzinfo=None)

        return datetime.now(timezone.utc)
```

```python
        utc_dt = datetime.now(timezone.utc)

        if not is_aware(dt):
            utc_dt = utc_dt.replace(tzinfo=None)

        return utc_dt
```